### ISSUE TITLE
[Feature] Use an already present puppeteer browser instance

### ISSUE THREAD

**Issue by Gummibeer** [webhintio/hint#3405]
# 🚀 Feature request

## Description

It would be great if the puppeteer connector accepts an already present puppeteer browser instance as argument. This way it doesn't have to launch its own instance. This is helpful if webhint is part of a larger project with more tools based on puppeteer.

## What scenarios will this solve?

I have an app based on puppeteer/chrome running lighthouse, taking screenshots and webhint and I want to let them run in the same chrome browser instance.

**Comment by molant** [comment]
Hi @Gummibeer,

we are using `puppeteer-core` that doesn't download the browser. If you use the [`browser` property](https://webhint.io/docs/user-guide/connectors/connector-puppeteer/#options) of the configuration it will try to find the given browser in your machine for known path locations. Alternative, you can pass the exact path to the browser to use via [`puppeteerOptions.executablePath`](https://pptr.dev/#?product=Puppeteer&version=master&show=api-puppeteerlaunchoptions)

It should look something like:

```json
{
  "connector": {
    "name": "puppeteer",
    "options": {
      "puppeteerOptions": {
        "executablePath": "pathToChromiumExecutable"
      }
    }
  }
}
```

Is this what you are looking for?

**Comment by Gummibeer** [comment]
No, I meant the already running instance. Like lighthouse accepts it. So I can create a chrome instance via chrome launcher, start a puppeteer via the launched chrome web debug port and pass the puppeteer instance to lighthouse, webhint and so on. This should also prevent the browser.info file in CWD.

**Comment by molant** [comment]
So you want something like [`puppeteer.connect`](https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-puppeteerconnectoptions) where you can pass `browserWSEndpoint` ?

You are using webhint's CLI, right? I assume you want this via `.hintrc` or do you have something else in mind?

**Comment by Gummibeer** [comment]
I use the nodejs package. So a programmatic approach would be good for me. A simple optional config key for example where I can pass the `browserWSEndpoint`.
